Godeps/_workspace/src/github.com/GoogleCloudPlatform/kubernetes/pkg/runtime/types.go

```diff
@@ -116,3 +116,17 @@ type Unknown struct {
 }
 
 func (*Unknown) IsAnAPIObject() {}
+
+// Unstructured allows objects that do not have Golang structs registered to be manipulated
+// generically. This can be used to deal with the API objects from a plug-in. Unstructured
+// objects still have functioning TypeMeta features-- kind, version, etc.
+// TODO: Make this object have easy access to field based accessors and settors for
+// metadata and field mutatation.
+type Unstructured struct {
```

Comment:
Unknown is much less expensive and is essentially "an opaque blob".  Not everyone who wants to use objects needs the processing Unstructured has.

----- Original Message -----
> > @@ -116,3 +116,17 @@ type Unknown struct {
> >  }
> >  
> >  func (*Unknown) IsAnAPIObject() {}
> > +
> > +// Unstructured allows objects that do not have Golang structs registered
> > to be manipulated
> > +// generically. This can be used to deal with the API objects from a
> > plug-in. Unstructured
> > +// objects still have functioning TypeMeta features-- kind, version, etc.
> > +// TODO: Make this object have easy access to field based accessors and
> > settors for
> > +// metadata and field mutatation.
> > +type Unstructured struct {
> 
> Why would we keep both `Unknown` and `Unstructured`?  Seems like
> `Unstructured` is a more convenient form that is equivalent in power.
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/openshift/origin/pull/1959/files#r29328825